proposals/stf-general-investment.md

```diff
@@ -0,0 +1,263 @@
+We expect this to have a compounding effect on overall productivity, as we estimate that more than 95% of the work is currently done by volunteers, many of whom are slowed down by lack of support and swift decision making.
+
+- Improve the contributor experience (300k)
+
+  Streamlining the contribution process should make it more attractive for newcomers to participate in development, and keep existing people around and motivated.
+  Growing the contributor community will make the ecosystem more robust against fluctuation.
+  We want to achieve this by making more systemic knowledge explicit and easier to discover, access, and manipulate, and thus ease onboarding of new contributors, and reduce contribution turnaround time.
+
+  - Tools for review automation (50k)
+  - Modularise the Nix configuration system (30k)
+  - Increase coverage and reduce duration of Nix integration testing (30k)
+  - Monitoring and data gathering for informed decision-making (30k)
+  - Professionalise the moderation team (50k)
+  - Organisation-wide declarative permission and notification management (60k)
+
+- Improve documentation (170k)
+
+  More focused, correct, up-to-date, and well-structured documentation will make the system more learnable.
+  This will over time reduce the number and size of blind spots in components that may otherwise end up unmaintained, a problem we are observing for many years now.
+  It should also tangibly reduce maintainer workload by making users and contributors more self-sufficient.
+
+  - Improve and speed up documentation rendering infrastructure (40k)
+  - Rework the Nixpkgs manual (60k)
+  - [Design and implement an information architecture](https://discourse.nixos.org/t/zurich-23-11-zhf-hackathon-and-ux-workshop-report/37848#ux-workshop-5) (50k)
+  - Build documentation testing infrastructure (20k)
+
+- Reduce ongoing costs for continuous integration and distribution (180k)
+
+  Infrastructure expenses are a signfificant risk for the ecosystem, because they are almost completely sponsored by companies who may end their support any time.
+  Reducing these ongoing costs will make it easier to migrate between providers or even self-host.
+  Improving CI performance will also contribute to faster workflows, and increase responsiveness to security incidents.
+  This will also impact all downstream users and significantly reduce their storage and energy requirements for the forseeable future.
+
+  - Deduplicate the cache storage (40k)
+  - Improve evaluation performance (100k)
+    - parallel evaluation
+    - evaluation cache
+    - memory layout optimisations
+    - minimal NixOS modules list
+  - Rewrite the Hydra front-end (40k)
```

Comment:
The motivation of rewriting the front-end is unclear to me.  Are there serious issues with the front-end?  Is the maintenance cost of the front-end too high?

Another question: why not rewrite the whole hydra instead of only the front-end?

Reply:
There are currently two severe issues with Hydra's backend:
- Compressing centrally does not scale
- The speed at which runnable jobs can be determined often does not saturate builders

Reply:
But here it only says that the front-end should be overhauled, not the backend? It should be clarified what exact part of Hydra is meant by this.

Reply:
```suggestion
  - Fix performance and maintenance bottlenecks in Hydra (40k)
```

Given that we have a 900 word limit and this is supposed to be the first of three iterations in the application process, it doesn't make much sense to go into any detail here. The people who would do the implementation work should elaborate on the details eventually.